### PR TITLE
Adding ts support for africastalking

### PR DESCRIPTION
**Pull request by james-kaguru**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

**Review by james-kaguru — COMMENTED** [review]
reviewed the extra changes

File: types/africastalking/package.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "private": true,
+    "type": "module"
```
Comment (jakebailey) [inline types/africastalking/package.json:3]: This package isn't a module; upstream is CJS and the d.ts file here says `export =`; you should delete this file.

File: types/africastalking/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (jakebailey) [inline types/africastalking/tsconfig.json:18]: This should be disabled; see the bot's comment.

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/africastalking/package.json (hunk 1 above)
Comment (James-Kaguru) [reply types/africastalking/package.json:3]: well noted @jakebailey

File: types/africastalking/tsconfig.json (hunk 2 above)
Comment (James-Kaguru) [reply types/africastalking/tsconfig.json]: already disabled them @jakebailey

Comment (jakebailey) [reply types/africastalking/tsconfig.json]: The file doesn't parse anymore. Can't have trailing commas in these files; DT is stricter here.

Comment (James-Kaguru) [reply types/africastalking/tsconfig.json]: already resolved..hopefully its now resolved

Comment (James-Kaguru) [reply types/africastalking/tsconfig.json]: unfortunately I cant seem to find the bots comment regarding what I should disable..for clarity purposes what in particular config was I to disable?

File: types/africastalking/africastalking-tests.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import AfricasTalking from "africastalking";
```
Comment (jakebailey) [inline types/africastalking/africastalking-tests.ts:1]:
```suggestion
import AfricasTalking = require("africastalking");
```

File: types/africastalking/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": false
```
Comment (jakebailey) [inline types/africastalking/tsconfig.json:18]:
```suggestion
```

File: types/africastalking/africastalking-tests.ts (hunk 3 above)
Comment (James-Kaguru) [reply types/africastalking/africastalking-tests.ts]: @jakebailey does the above change mean that even when the client uses this library they will have to import it that way?

Comment (jakebailey) [reply types/africastalking/africastalking-tests.ts]: No, just that DT is very strict about getting at the types of libraries; if someone actually was using the esModuleInterop setting, this issue would be smoothed over, more or less.

**Review by jakebailey — APPROVED** [review]

**Comment by James-Kaguru** [comment]
Ready to merge

**Pull request merged**